tests/fault_tolerance/hardware/fault-injection-service/helpers/gpu_discovery.py

```
@@ -0,0 +1,272 @@
+        gpu_ids = []
+        for line in result.stdout.decode().splitlines():
+            line = line.strip()
+            if line.isdigit():
+                gpu_ids.append(int(line))
+        
+        if not gpu_ids:
+            logger.warning(f"No GPUs found in pod {pod.name}")
+            return []
+        
+        logger.debug(f"Available GPU IDs in pod {pod.name}: {gpu_ids}")
+        return gpu_ids
```

Ruff (0.14.7)

Consider moving this statement to an `else` block

(TRY300)